pip/vcs/git.py

```diff
@@ -134,7 +134,7 @@ def obtain(self, dest):
             rev_options = [rev]
             rev_display = ' (to %s)' % rev
         else:
-            rev_options = ['origin/master']
+            rev_options = ['refs/remotes/origin/HEAD']
```

Comment:
If this is a bug in `vcs/git.py`, it seems like it'd be worth adding a lower-level / finer-grained test inside `test_install_vcs_git.py` instead of only checking via a high-level end-to-end test. A possible example to compare to is `test_obtain_should_recognize_auth_info_url()`.